core/src/main/java/com/statful/client/core/buffer/AggregatedBuffer.java

```diff
@@ -63,7 +65,19 @@ public final boolean addToBuffer(final String metric, final Aggregation aggregat
         aggregatedBuffer.put(aggregationFrequency.toString(), aggregatedFreqBuffer);
         buffer.put(aggregation.toString(), aggregatedBuffer);
 
-        return aggregatedFreqBuffer.offer(metric);
+        boolean inserted = aggregatedFreqBuffer.offer(metric);
+        if (!inserted) {
+            try {
+                LOGGER.fine("Buffer is full, trying to discard oldest metric.");
+                aggregatedFreqBuffer.remove();
```

Comment:
Instead of remove we could use pool ( https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ArrayBlockingQueue.html#poll() ) which would allow us to log the discarded metric if we wanted, which I think we want, and doesn't throw exception if it's empty - returns null).

Thinking out loud, should we drop more then one metric?